### DISCUSSION TITLE
Is it not possible to include a reordable with a BelongsToMany?

### DISCUSSION THREAD

**Discussion by SebastianUhlig** [filamentphp/filament#6422] [Help] [answered]
Question is already in the title. I am currently trying to build a relationship table.

In one resource you are able to create menu items, in a another you can attach the created menu items.

So far it works fine, but I cannot reorder in the RelationManager, where I have attached the menu items.

I can add the method `->reorderable()` in the RelationManager table and also activate and drag it, but as soon as I moved it, it wants to change the 'order' of the parent "menu items" and not the belongsToMany-Table.

Did anyone have an idea?

**Comment by danharrin** [comment]
Sounds like it could be a bug, so try and fix it and open a PR!

**Reply by SebastianUhlig** [reply]
> Sounds like it could be a bug, so try and fix it and open a PR!

Well, it could be, but i'm not sure if it is one. And if it's a bug, I don't know immediately how to fix it 😅
Should it even be possible to use a reorderable for a BelongsToMany table/relation?

**Reply by danharrin** [reply]
Probably yes! I'm not expecting you to immediately know how to fix it, but you can have a look into it and try and add support for saving to the pivot table! It's probably not different from how youd approach it in Laravel outside of Filament

**Reply by SebastianUhlig** [reply]
Sorry, the previous answer was complete bullshit because I changed something in the code that did not originate...

But it's true, if I remove line 25 completely from the if statement, it works. Both with normal reorderables and with my BelongsToMany.
https://github.com/filamentphp/filament/blob/2.x/packages/tables/src/Concerns/CanReorderRecords.php#L25

But now again: What is the deep meaning behind this statement at this line? Why is the array, what you get in `$relationship->getPivotColumns()` empty and therefore false, if you want to reorder the table?

**Reply by danharrin** [reply, marked as answer]
Do you have `withPivot('order')` defined on the relationship?

**Reply by SebastianUhlig** [reply]
> Do you have `withPivot('order')` defined on the relationship?

Okay, that solves my problem. I completely forgot to set it xD
Thanks and sorry for my mistake
